### ISSUE TITLE
[Feature request] Ignore Dart daylight savings

### ISSUE THREAD

**Issue by JCKodel** [xantiagoma/dart_date#5]
This:

```dart
final date = DateTime(2017, 1, 1);

print(date.addDays(1451));
```

gives as response `2020-12-21 23:00:00.000`, because somehow Dart considers daylight savings (which is VERY dangerous, for example: Brazil abolished daylight savings in 2019. If it is reintroduced in the future, current apps will never take that into consideration).

This:

```dart
final date = DateTime(2017, 1, 1);

print(DateTime(date.year, date.month, date.day + 1451));
```

surprisingly gives the "right" answer: `2020-12-22 00:00:00.000`.

It would be nice if the `addXXX()` and `subXXX()` methods could take a bool flag to choose what kind of calculation we want.

For example, in the `addDays` functions we have:

```dart
/// Add a certain amount of days to this date
DateTime addDays(int amount) => add(Duration(days: amount));
```

We could have this:

```dart
/// Add a certain amount of days to this date
DateTime addDays(int amount, [bool ignoreDaylightSavings = false]) {
  if(ignoreDaylightSavings) {
    return DateTime(year, month, day + amount);
  } else {
    return add(Duration(days: amount));
  }
}
```

I think all `DateTime` operations that uses `add(Duration())` will have this "feature".

**Comment by JCKodel** [comment]
Done in https://github.com/xantiagoma/dart_date/pull/6 (but I changed some minor things, to comply with analysis options).

**Comment by xantiagoma** [comment]
Hi @JCKodel  Doesn't it work to handle the dates as UTC to avoid the problem you mention with daylight saving time? I think what you mention depends on the local configuration of the device and if in its timezone it is needed or not.

**Comment by JCKodel** [comment]
> Hi @JCKodel Doesn't it work to handle the dates as UTC to avoid the problem you mention with daylight saving time? I think what you mention depends on the local configuration of the device and if in its timezone it is needed or not.

`.toUtc()` still calculates wrong (because at some point I need to convert it back to local).

**Comment by xantiagoma** [comment]
Merged #6 
Published in https://pub.dev/packages/dart_date/changelog#109
